### PR TITLE
Add 'Documenting Code' to contribution policy

### PR DESCRIPTION
**Pull request by hackmastera**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -38,6 +40,24 @@ You should also add yourself to the `CONTRIBUTORS.md` file in the root of the pr
   * Then checkout the new branch with `git checkout fix/master/my_contribution`.
   * Please avoid working directly on the `master` branch.
   * You may find the [hub suite of commands](https://github.com/defunkt/hub) helpful
+* Make sure you have added the necessary tests for your changes.
```
Comment (mjgiarlo) [inline CONTRIBUTING.md]: How about "Make sure you have added sufficient tests and documentation for your changes." to reinforce what's expounded upon below?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -38,6 +40,24 @@ You should also add yourself to the `CONTRIBUTORS.md` file in the root of the pr
...
+* Run _all_ the tests to assure nothing else was accidentally broken.
+
+### Documenting Code
+
+* All new public methods, modules, and classes should include inline documentation in [YARD](http://yardoc.org/).
+  * Documentation should seek to answer the question "why does this code exist?"
+* Document private / protected methods as desired.
+* If you are working in a file with no prior documentation, do try to document as you gain understanding of the code.
+  * If you don't know exactly what a bit of code does, it is extra likely that it needs to be documented. Take a stab at it and ask for feedback in your pull request.
```
Comment (mjgiarlo) [inline CONTRIBUTING.md]: And folks can also use the "blame" button on GitHub to figure out who originally wrote the code in question, so they can ping that person in the PR to ask for clarification.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -38,6 +40,24 @@ You should also add yourself to the `CONTRIBUTORS.md` file in the root of the pr
...
+  * This work greatly increases the usability of the code base and supports the on-ramping of new committers.
+  * We will all be understanding of one anothers' time constraints in this area.
```
Comment (mjgiarlo) [inline CONTRIBUTING.md]: "one another's", I think? My inner "Mrs. D from 12th grade" backs me up but my memory is hazy at times. ;)

**Comment by hackmastera** [comment]
Thanks for the suggestions, @mjgiarlo, these have been incorporated. Not sure what the next step is -- should this language be approved on a tech call before merging?

**Comment by awead** [comment]
@hackmastera Thanks for doing this! :clap:  I'd post a note to hydra-tech and give folks a week to comment or raise objections. After that, one of us can just merge it. I'm not against talking it over on a tech call, but we've already done that and I think most of us are on-board with the changes.

**Comment by mjgiarlo** [comment]
:+1: to @awead's remarks

**Comment by dchandekstark** [comment]
:+1: :clap:

**Pull request merged**